source/configuration_manual/ssl_configuration.rst

```diff
@@ -48,3 +48,18 @@ If you need different SSL certificates for IMAP and POP3 protocols, you can put
 Dovecot supports also using TLS SNI extension for giving different SSL certificates based on the server name when using only a single IP address, but the TLS SNI isn't yet supported by all clients so that may not be very useful. 
 
 It's anyway possible to configure it by using ``local_name imap.example.com {}`` blocks.
+
+.. _ssl_ja3:
+
+JA3 identifier
+--------------
+
+.. versionadded:: v2.4.0;v3.0.0
+
+Dovecot supports calculating `JA3 hash <https://engineering.salesforce.com/tls-fingerprinting-with-ja3-and-ja3s-247362855967/>`__ for checking client TLS implementation.
+This adds ``ssl_ja3`` and ``ssl_ja3_hash`` to :ref:`login variables <variables-login>`, to be used with :dovecot_core:ref:`login_log_format_elements`
+and ``ssl_ja3_hash`` for :ref:`authentication variables <variables-auth>`, to be used with e.g. :ref:`authentication-auth_policy`.
+
+Please note that since IMAP does not usually support ALPN or NPN extension. This makes hash databases used with http protocol are generally not suitable sources for values.
```

Comment:
* "since" word too much
 * "are" word too much
 * is it "IMAP clients" that don't support? It's not IMAP protocol itself, right?
 *

Reply:
Or is it that there is no ALPN/NPN standard for IMAP protocol? In that case I'd say "Please note that since ALPN or NPN extensions haven't been standardized for IMAP yet, IMAP clients usually use them."

Reply:
Well, there is actually. I requested the registration of POP3, IMAP, LMTP and ManageSieve years ago, we just never got round implementing this into Dovecot properly. 

It would make a lot of sense to do ALPN/NPN support so that client can say that "Hey, after TLS I expect to speak IMAP" when connecting to IMAP port, so we could already tell the client that "no, this is not IMAP port". Doing this in login-client would be trivial.